src/js/workshop-comments.js

```diff
@@ -0,0 +1,132 @@
+/* global convertDate */
+
+const commentForm = document.getElementById("comment-form");
+
+commentForm.addEventListener("submit", (event) => {
+	event.preventDefault();
+
+	const nameDomElement = document.getElementById("name");
+	const name = nameDomElement.value.trim();
+	const commentDomElement = document.getElementById("comment");
+	const comment = commentDomElement.value.trim();
+
+	const submitFailureMessage = document.querySelector(".submit-failure-message");
+
+	// Hide required and comment submission message.
+	// Note: It's necessary to hide "required" and "succesful/failed" submitted comment message indicators programmatically even though they have been hidden initially by CSS
+	// in the case the same user submits more comments after submitting an initial comment.
+	document.querySelectorAll(".required, .submit-success-message, .submit-failure-message").forEach(element => {
+		applyVisibility(element, false);
+	});
+
+	// Verify input Fields.
+	let isValueMissing = false;
+
+	if (name === "") {
+		applyVisibility(document.querySelector(".required-name"), true, "inline");
+		isValueMissing = true;
+	}
+
+	if (comment === "") {
+		applyVisibility(document.querySelector(".required-comment"), true, "inline");
+		isValueMissing = true;
+	}
+
+	if (isValueMissing) {
+		return;
+	}
+
+	// Disable input fields when proccessing a new request.
+	disableFields(true);
+
+	const XHR = new XMLHttpRequest();
+
+	// Bind the FormData object and the form element.
+	const commentFormData = new FormData( commentForm );
+
+	// Submit and add comment to top of comment section of page on successful data submission.
+	XHR.addEventListener( "load", function() {
+		if (XHR.status === 200) {
+			showSubmittedComment(nameDomElement, commentDomElement);
```

Comment:
Passing in `name` and `comment` as arguments rather than their DOM elements. The clean-up of these values have already been performed here and don't need to do the same in the `showSubmittedComment()` function.